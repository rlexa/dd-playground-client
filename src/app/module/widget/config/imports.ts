import { CommonModule } from '@angular/common';
import { MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatCardModule, MatListModule];
