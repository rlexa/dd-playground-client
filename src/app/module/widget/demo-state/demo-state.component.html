<div *ngIf="state$ | async as state" ddFlexbox="vyt">
  <mat-card>
    <mat-card-title>State (PARSED)</mat-card-title>
    <mat-card-content>
      <app-simple-view [data]="state" [isWidthConstrained]="false" [isDense]="true"></app-simple-view>
    </mat-card-content>
  </mat-card>
</div>