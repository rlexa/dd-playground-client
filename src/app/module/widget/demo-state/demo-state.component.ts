import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
import { RxStateService } from 'app/rx-state';
import { DoneSubject, RxCleanup } from 'dd-rxjs';

@Component({
  selector: 'app-demo-state',
  templateUrl: './demo-state.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class DemoStateComponent implements OnDestroy {
  constructor(private rxState: RxStateService) { }
  @RxCleanup() private readonly done$ = new DoneSubject();
  readonly state$ = this.rxState.watch(state => state, this.done$);
  ngOnDestroy() { }
}
