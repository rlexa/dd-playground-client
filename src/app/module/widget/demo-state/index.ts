export * from './demo-state.component';
export * from './demo-state.module';
