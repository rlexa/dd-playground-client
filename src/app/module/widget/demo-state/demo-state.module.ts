import {NgModule} from '@angular/core';
import {RouterModule, Routes} from '@angular/router';
import {ROUTE_ROOT, ROUTE_WILDCARD} from 'app/routing';
import {DemoStateComponent} from './demo-state.component';
import {imports} from './imports';

const ROUTING: Routes = [
  {path: ROUTE_ROOT, component: DemoStateComponent},
  {path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT, pathMatch: 'full'},
];

@NgModule({imports: [...imports, RouterModule.forChild(ROUTING)], exports: [DemoStateComponent], declarations: [DemoStateComponent]})
export class DemoStateModule {}
