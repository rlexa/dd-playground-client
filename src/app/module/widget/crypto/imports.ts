import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import {
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
} from '@angular/material';
import { RouterModule } from '@angular/router';
import { CryptoApiModule } from 'app/module/service/crypto-api';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  FormsModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
  FlexboxModule,
  CryptoApiModule,
  SimpleViewModule,
];
