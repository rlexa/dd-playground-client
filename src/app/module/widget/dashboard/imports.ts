import { CommonModule } from '@angular/common';
import { MatButtonModule, MatToolbarModule, MatTooltipModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { VersionModule } from 'app/module/widget/version';
import { WidgetsModule } from 'app/widgets/widgets.module';
import { FlexboxModule } from 'dd-flexbox';
import 'hammerjs';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatToolbarModule, MatTooltipModule, FlexboxModule,
  IconPipeModule, VersionModule,
  // remove
  RoutedContentModule, WidgetsModule,
];
