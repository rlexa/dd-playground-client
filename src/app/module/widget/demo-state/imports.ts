import { CommonModule } from '@angular/common';
import { MatCardModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'dd-flexbox';

export const imports = [CommonModule, RouterModule, MatCardModule, FlexboxModule, SimpleViewModule];
