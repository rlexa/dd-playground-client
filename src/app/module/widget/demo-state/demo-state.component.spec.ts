import {async, TestBed} from '@angular/core/testing';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';
import {RouterTestingModule} from '@angular/router/testing';
import {provideRxState} from 'src/app/rx-state/test';
import {DemoStateComponent} from './demo-state.component';
import {imports} from './imports';

describe('DemoStateComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [NoopAnimationsModule, RouterTestingModule, ...imports],
      declarations: [DemoStateComponent],
      providers: [...provideRxState],
    }).compileComponents();
  }));

  test('is created', () => {
    const fixture = TestBed.createComponent(DemoStateComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
