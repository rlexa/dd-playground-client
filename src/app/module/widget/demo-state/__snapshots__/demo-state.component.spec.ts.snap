// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoStateComponent is created 1`] = `
<app-demo-state
  done$={[Function DoneSubject]}
  rxState={[Function RxStateService]}
  state$={[Function BehaviorSubject]}
>
  <div
    class="ng-star-inserted"
    ddflexbox="vyt"
    style="display: flex; flex-direction: column; justify-content: flex-start; align-items: stretch; align-content: stretch; flex-wrap: no-wrap;"
  >
    <mat-card
      class="mat-card _mat-animation-noopable"
    >
      <mat-card-title
        class="mat-card-title"
      >
        State (PARSED)
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <app-simple-view
          class="ng-tns-c1-0"
        >
          
          <div
            class="position-relative ng-tns-c1-0 ng-star-inserted"
          >
            <div
              class="scroll-x-auto"
              ddflexbox="hys"
              style="max-width: 100%; display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
            >
              
              
              <mat-list
                class="mat-list-auto-height mat-list mat-list-base ng-tns-c1-0 ng-star-inserted"
                dense="true"
              >
                
                
                
                
                
                
                
                <table
                  class="ng-tns-c1-0 ng-star-inserted"
                >
                  
                  <tr
                    class="ng-tns-c1-0 ng-star-inserted"
                  >
                    <td
                      class="ng-tns-c1-0"
                      style="vertical-align: top;"
                    >
                      <mat-list-item
                        class="mat-list-item"
                      >
                        <div
                          class="mat-list-item-content"
                        >
                          <div
                            class="mat-list-item-ripple mat-ripple"
                            mat-ripple=""
                          />
                          <div
                            class="mat-list-text"
                          />
                          <span
                            class="no-break-text ng-tns-c1-0"
                          >
                             Game 
                          </span>
                        </div>
                      </mat-list-item>
                    </td>
                    <td
                      class="ng-tns-c1-0"
                    >
                      <mat-list-item
                        class="mat-list-item ng-trigger ng-trigger-TRIGGER_WobbleX ng-animating"
                        style=""
                      >
                        <div
                          class="mat-list-item-content"
                        >
                          <div
                            class="mat-list-item-ripple mat-ripple"
                            mat-ripple=""
                          />
                          <div
                            class="mat-list-text"
                          />
                          <span
                            class="break-text ng-tns-c1-0"
                          >
                            
                            
                            
                            
                            
                            
                            
                            <app-simple-view
                              class="ng-tns-c1-1 ng-tns-c1-0 ng-star-inserted"
                              style=""
                            >
                              
                              <div
                                class="position-relative ng-tns-c1-1 ng-star-inserted"
                              >
                                <div
                                  class="scroll-x-auto"
                                  ddflexbox="hys"
                                  style="max-width: 100%; display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
                                >
                                  
                                  <button
                                    class="inline-button mat-button mat-button-base ng-tns-c1-1 _mat-animation-noopable ng-star-inserted"
                                    mat-button=""
                                  >
                                    <span
                                      class="mat-button-wrapper"
                                    >
                                      <i
                                        class="material-icons rot-90 ng-tns-c1-1"
                                      >
                                        expand_more
                                      </i>
                                    </span>
                                    <div
                                      class="mat-button-ripple mat-ripple"
                                      matripple=""
                                    />
                                    <div
                                      class="mat-button-focus-overlay"
                                    />
                                  </button>
                                  
                                </div>
                              </div>
                            </app-simple-view>
                            
                          </span>
                        </div>
                      </mat-list-item>
                    </td>
                  </tr>
                  <tr
                    class="ng-tns-c1-0 ng-star-inserted"
                  >
                    <td
                      class="ng-tns-c1-0"
                      style="vertical-align: top;"
                    >
                      <mat-list-item
                        class="mat-list-item"
                      >
                        <div
                          class="mat-list-item-content"
                        >
                          <div
                            class="mat-list-item-ripple mat-ripple"
                            mat-ripple=""
                          />
                          <div
                            class="mat-list-text"
                          />
                          <span
                            class="no-break-text ng-tns-c1-0"
                          >
                             Global Values 
                          </span>
                        </div>
                      </mat-list-item>
                    </td>
                    <td
                      class="ng-tns-c1-0"
                    >
                      <mat-list-item
                        class="mat-list-item ng-trigger ng-trigger-TRIGGER_WobbleX ng-animating"
                        style=""
                      >
                        <div
                          class="mat-list-item-content"
                        >
                          <div
                            class="mat-list-item-ripple mat-ripple"
                            mat-ripple=""
                          />
                          <div
                            class="mat-list-text"
                          />
                          <span
                            class="break-text ng-tns-c1-0"
                          >
                            
                            
                            
                            
                            
                            
                            
                            <app-simple-view
                              class="ng-tns-c1-2 ng-tns-c1-0 ng-star-inserted"
                              style=""
                            >
                              
                              <div
                                class="position-relative ng-tns-c1-2 ng-star-inserted"
                              >
                                <div
                                  class="scroll-x-auto"
                                  ddflexbox="hys"
                                  style="max-width: 100%; display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
                                >
                                  
                                  <button
                                    class="inline-button mat-button mat-button-base ng-tns-c1-2 _mat-animation-noopable ng-star-inserted"
                                    mat-button=""
                                  >
                                    <span
                                      class="mat-button-wrapper"
                                    >
                                      <i
                                        class="material-icons rot-90 ng-tns-c1-2"
                                      >
                                        expand_more
                                      </i>
                                    </span>
                                    <div
                                      class="mat-button-ripple mat-ripple"
                                      matripple=""
                                    />
                                    <div
                                      class="mat-button-focus-overlay"
                                    />
                                  </button>
                                  
                                </div>
                              </div>
                            </app-simple-view>
                            
                          </span>
                        </div>
                      </mat-list-item>
                    </td>
                  </tr>
                  <tr
                    class="ng-tns-c1-0 ng-star-inserted"
                  >
                    <td
                      class="ng-tns-c1-0"
                      style="vertical-align: top;"
                    >
                      <mat-list-item
                        class="mat-list-item"
                      >
                        <div
                          class="mat-list-item-content"
                        >
                          <div
                            class="mat-list-item-ripple mat-ripple"
                            mat-ripple=""
                          />
                          <div
                            class="mat-list-text"
                          />
                          <span
                            class="no-break-text ng-tns-c1-0"
                          >
                             Ui 
                          </span>
                        </div>
                      </mat-list-item>
                    </td>
                    <td
                      class="ng-tns-c1-0"
                    >
                      <mat-list-item
                        class="mat-list-item ng-trigger ng-trigger-TRIGGER_WobbleX ng-animating"
                        style=""
                      >
                        <div
                          class="mat-list-item-content"
                        >
                          <div
                            class="mat-list-item-ripple mat-ripple"
                            mat-ripple=""
                          />
                          <div
                            class="mat-list-text"
                          />
                          <span
                            class="break-text ng-tns-c1-0"
                          >
                            
                            
                            
                            
                            
                            
                            
                            <app-simple-view
                              class="ng-tns-c1-3 ng-tns-c1-0 ng-star-inserted"
                              style=""
                            >
                              
                              <div
                                class="position-relative ng-tns-c1-3 ng-star-inserted"
                              >
                                <div
                                  class="scroll-x-auto"
                                  ddflexbox="hys"
                                  style="max-width: 100%; display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
                                >
                                  
                                  <button
                                    class="inline-button mat-button mat-button-base ng-tns-c1-3 _mat-animation-noopable ng-star-inserted"
                                    mat-button=""
                                  >
                                    <span
                                      class="mat-button-wrapper"
                                    >
                                      <i
                                        class="material-icons rot-90 ng-tns-c1-3"
                                      >
                                        expand_more
                                      </i>
                                    </span>
                                    <div
                                      class="mat-button-ripple mat-ripple"
                                      matripple=""
                                    />
                                    <div
                                      class="mat-button-focus-overlay"
                                    />
                                  </button>
                                  
                                </div>
                              </div>
                            </app-simple-view>
                            
                          </span>
                        </div>
                      </mat-list-item>
                    </td>
                  </tr>
                </table>
                
                
              </mat-list>
            </div>
          </div>
        </app-simple-view>
      </mat-card-content>
    </mat-card>
  </div>
</app-demo-state>
`;
